sounddevice.py

```diff
@@ -2289,14 +2289,41 @@ def reset(self):
         vars(self).clear()
         self.__init__()
 
+
 if not hasattr(_ffi, 'I_AM_FAKE'):
     # This object shadows the 'default' class, except when building the docs.
     default = default()
 
 
 class PortAudioError(Exception):
-    """This exception will be raised on PortAudio errors."""
+    """This exception will be raised on PortAudio errors.
+    
+    Attributes
+    ----------
+    args
+        A variable length tuple containing the following parameters when
+        available:
+
+        error_message
+            The user error message.
+        pa_error_code
+            The PortAudio `PaErrorCode` value.
+        hosterror_info
+            A 3-tuple containing the host API index, host error code, and
+            the host error message or a zero length string if no message is
+            available.
 
+    """
+    def __str__(self):
+        errormsg = self.args[0] if len(self.args) else ''
+        if len(self.args) > 1:
+            errormsg = "{} [PaErrorCode {}]".format(errormsg, self.args[1])
+        if len(self.args) > 2:
+            host_api, hosterror_code, hosterror_text = self.args[2]
+            hostname = query_hostapis(host_api)['name']
+            errormsg = "{}: '{}' [{} error {}]".format(errormsg, hosterror_text,
```

Comment:
Again this is a coding habit. '{1}' is more explicit than assuming {1!r} will add quotes around the object. And in fact with Python2 this will cause a `u` (unicode string) to be prepended to the error message which I don't care for (but maybe this is what you intended?).